data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -168,6 +169,24 @@ else if (isEventNameCreated(entry.getValue().get(0))) {
         return deleteMessageBatchRequestEntryCollection;
     }
 
+    private Optional<DeleteMessageBatchRequestEntry> processS3Object(
+            final Map.Entry<Message, List<S3EventNotification.S3EventNotificationRecord>> entry,
+            final S3ObjectReference s3ObjectReference) {
+        Optional<DeleteMessageBatchRequestEntry> deleteMessageBatchRequestEntry = Optional.empty();
+        // SQS messages won't be deleted if we are unable to process S3Objects because of S3Exception: Access Denied
+        try {
+            s3Service.addS3Object(s3ObjectReference);
+            deleteMessageBatchRequestEntry = Optional.of(buildDeleteMessageBatchRequestEntry(entry.getKey()));
+            sqsMessageDelayTimer.record(Duration.between(
+                    Instant.ofEpochMilli(entry.getValue().get(0).getEventTime().toInstant().getMillis()),
+                    Instant.now()
+            ));
+        } catch (final Exception e) {
+            LOG.warn("Unable to process S3Object: s3ObjectReference={}.", s3ObjectReference, e);
```

Comment:
It might be outside the scope of this PR but it would be nice to expose the number of objects that threw exceptions as a metric.

Reply:
We do catch these metrics further down.